TROUBLESHOOTING.md

```diff
@@ -0,0 +1,89 @@
+
+**Table of Contents**
+1.	[Four triage steps](#four-triage-steps)
+2.	[Understanding the cause of an issue](#understand-the-cause-of-an-issue)
+3.	[Configuration issues](#configuration-issues)
+4.	[Bundler/Framework issues](#bundlerframework-issues)
+5.	[JS Maps SDK issues](#maps-sdk-issues)
+
+## Four triage steps
+
+There are four basic steps you can take to help isolate problems.
+
+**Step 1 – Review.** Examine the error message and the entire error dump. Most of the time there are clues in the error that will help solve the problem. Take note of when the error happens. Some errors only occur during the build process, others may only show up when you run the app in the browser, or only when the app is deployed to the production environment.
+
+**Step 2 - Search.** Search for error message on the [Esri Community site](https://community.esri.com/t5/arcgis-maps-sdk-for-javascript/ct-p/arcgis-api-for-javascript) or the internet, and research what might be causing the issue. This is often the fastest and easiest way to look for answers or get hints on how to solve a problem. Depending on the error, you should consider reviewing the JS Maps SDK release notes for [breaking changes](https://developers.arcgis.com/javascript/latest/release-notes/#breaking-changes) or [bug fixes and enhancements](https://developers.arcgis.com/javascript/latest/release-notes/#bug-fixes-and-enhancements).
+
+**Step 3 - Reproduce.** Try to reproduce the issue in a simple, focused stand-alone HTML/JavaScript app. The [ES module samples](./esm-samples/) in this repository are a great starting point. This will help determine if the issue is related to the JS Maps SDK, or not. Sometimes, in the process of trying to reproduce the error you will solve the issue.
+
+**Step 4 – Create a sample.** Always try to provide a simple sample when posting your question to the Esri Community Site. There are great online options for sharing samples including [codepen](https://codepen.io/), [github.com](https://github.com/), [codesandbox](https://codesandbox.io/) or [StackBlitz](https://stackblitz.com/). Code snippets often do not provide enough information about a project because they don’t include information on configuration, dependencies or coding patterns. Providing a working sample is essential to allow others to quickly reproduce your issue. Sometimes the process of creating the sample helps solve the problem.
+
+## Understand the cause of an issue
+
+With all the various parts in modern build systems, determining where the problem is occurring is critical. There are typically three categories where problems occur: configuration, bundler/framework, or the JS Maps SDK.
+
+## Configuration issues
```

Comment:
Empty newline after header items, couple cases here. Do newlines matter if header 2 or 3?

Reply:
Ah yes, I can remove them. Github markdown ignores them, but I'll fix them for consistency.